website/server/models/group.js

```diff
@@ -344,6 +344,8 @@ schema.statics.toJSONCleanChat = async function groupToJSONCleanChat (group, use
     await getGroupChat(group);
   }
 
+  group.translateSystemMessages(user);
+
   let toJSON = group.toJSON();
```

Comment:
I've lost the previous conversation but going back to this: I know that the group object is not going to be saved after this but it's bad practice to apply changes that will never end up in the db to a db document class and across the entire project we first serialize to a plain js object and then change what's needed

Reply:
So the class method is not needed anymore... Some wasted work and time detected.

Reply:
Well you can still use it and have it return the serialized data

Reply:
Changed back to static method, and now it translate on the serialized object. ab0dae8df359ae64d083b97e54d1e9bb4f93e95d

Hope you're done with this kind of reviews, I think I've already worked waaay too much in this, more than I should.

If there are any other improvements you find, please do it after merging this and in a new PR.

Reply:
> Hope you're done with this kind of reviews, I think I've already worked waaay too much in this, more than I should.

As I said previously we're sorry it took so long with this PR and if you don't want to work on it anymore that's fine and someone from the staff will finish it but I'm not going to merge it if there are still issues, I'll give it another look later but at least there's the problem with the preferences object being undefined in some case

Reply:
> someone from the staff will finish it

The preferences problem is the only thing missing. If I change the `lang` variable manually, everything works as expected. There is no missing feature here.

And it is not about me working on it or not. I only want this PR to be merged. That's why I created it, after all.

When I said "this kind of reviews", I meant refactorings you ask. "_It should be implemented this way and not that way_" or "_It is better this way_". I really think that this is not the kind of thing you should write in a 1 year 3 months old PR where the owner changed EVERYTHING he was asked to do so far, and always within a week.

Lets focus on creating a working feature. There is no need to merge it only then it becomes 100% _perfect_. Improvements and refactorings can be done later, by the Habitica team, as you said. But on another PR, please.